### PR TITLE
migrate kubelet/cm/container logs to structured logging

### PR DESCRIPTION
**Pull request by shiyajuan123**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
Migrate `pkg/kubelet/cm/container_manager_linux.go` and `pkg/kubelet/cm/container_manager_stub.go` to structured logging
#### Which issue(s) this PR fixes:
Ref #98976

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/kubelet/cm/container_manager_linux.go [hunk 1]
```diff
@@ -310,12 +310,12 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 			return nil, err
 		}
 
-		klog.Infof("[topologymanager] Initializing Topology Manager with %s policy and %s-level scope", nodeConfig.ExperimentalTopologyManagerPolicy, nodeConfig.ExperimentalTopologyManagerScope)
+		klog.InfoS("[Topologymanager] Initializing Topology Manager with policy and level scope", "policy", nodeConfig.ExperimentalTopologyManagerPolicy, "scope", nodeConfig.ExperimentalTopologyManagerScope)
```
Comment (yuzhiquan) [inline pkg/kubelet/cm/container_manager_linux.go:313]:
```
klog.InfoS("[Topologymanager] Initializing Topology Manager", "policy", nodeConfig.ExperimentalTopologyManagerPolicy, "scope-level", nodeConfig.ExperimentalTopologyManagerScope)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 2]
```diff
@@ -872,7 +872,7 @@ func getPidsForProcess(name, pidFile string) ([]int, error) {
 
 	// Return error from getPidFromPidFile since that should have worked
 	// and is the real source of the problem.
-	klog.V(4).Infof("unable to get pid from %s: %v", pidFile, err)
+	klog.V(4).InfoS("Unable to get pid from", "pidFile", pidFile, "err", err)
```
Comment (shubheksha) [inline pkg/kubelet/cm/container_manager_linux.go:875]:
```suggestion
	klog.V(4).InfoS("Unable to get pid from", "pidFilePath", pidFile, "err", err)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 3]
```diff
@@ -222,7 +222,7 @@ func NewContainerManager(mountUtil mount.Interface, cadvisorInterface cadvisor.I
 		swapData, err := ioutil.ReadFile(swapFile)
 		if err != nil {
 			if os.IsNotExist(err) {
-				klog.Warningf("file %v does not exist, assuming that swap is disabled", swapFile)
+				klog.InfoS("File does not exist, assuming that swap is disabled", "file", swapFile)
```
Comment (shubheksha) [inline pkg/kubelet/cm/container_manager_linux.go:225]:
```suggestion
				klog.InfoS("File does not exist, assuming that swap is disabled", "filePath", swapFile)
```

Comment (ehashman) [reply pkg/kubelet/cm/container_manager_linux.go:225]: Should just be path

```suggestion
				klog.InfoS("File does not exist, assuming that swap is disabled", "path", swapFile)
```

File: pkg/kubelet/cm/container_manager_linux.go (hunk 1 above)
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:313]:
```suggestion
		klog.InfoS("Initializing Topology Manager", "policy", nodeConfig.ExperimentalTopologyManagerPolicy, "scope", nodeConfig.ExperimentalTopologyManagerScope)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 4]
```diff
@@ -449,9 +449,9 @@ func setupKernelTunables(option KernelTunableBehavior) error {
 		case KernelTunableError:
 			errList = append(errList, fmt.Errorf("invalid kernel flag: %v, expected value: %v, actual value: %v", flag, expectedValue, val))
 		case KernelTunableWarn:
-			klog.V(2).Infof("Invalid kernel flag: %v, expected value: %v, actual value: %v", flag, expectedValue, val)
+			klog.V(2).InfoS("Invalid kernel flag, expected value, actual value", "flag", flag, "expectedValue", expectedValue, "actualValue", val)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:452]:
```suggestion
			klog.V(2).InfoS("Invalid kernel flag", "flag", flag, "expectedValue", expectedValue, "actualValue", val)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 5]
```diff
@@ -449,9 +449,9 @@ func setupKernelTunables(option KernelTunableBehavior) error {
...
 		case KernelTunableModify:
-			klog.V(2).Infof("Updating kernel flag: %v, expected value: %v, actual value: %v", flag, expectedValue, val)
+			klog.V(2).InfoS("Updating kernel flag, expected value, actual value", "flag", flag, "expectedValue", expectedValue, "actualValue", val)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:454]:
```suggestion
			klog.V(2).InfoS("Updating kernel flag", "flag", flag, "expectedValue", expectedValue, "actualValue", val)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 6]
```diff
@@ -500,13 +500,13 @@ func (cm *containerManagerImpl) setupNode(activePods ActivePodsFunc) error {
 		// Check the cgroup for docker periodically, so kubelet can serve stats for the docker runtime.
 		// TODO(KEP#866): remove special processing for CRI "docker" enablement
 		cm.periodicTasks = append(cm.periodicTasks, func() {
-			klog.V(4).Infof("[ContainerManager]: Adding periodic tasks for docker CRI integration")
+			klog.V(4).InfoS("[ContainerManager]: Adding periodic tasks for docker CRI integration")
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:503]:
```suggestion
			klog.V(4).InfoS("Adding periodic tasks for docker CRI integration")
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 7]
```diff
@@ -500,13 +500,13 @@ func (cm *containerManagerImpl) setupNode(activePods ActivePodsFunc) error {
...
 			cont, err := getContainerNameForProcess(dockerProcessName, dockerPidFile)
 			if err != nil {
-				klog.Error(err)
+				klog.ErrorS(err, "Failed to get container name for process")
 				return
 			}
-			klog.V(2).Infof("[ContainerManager]: Discovered runtime cgroups name: %s", cont)
+			klog.V(2).InfoS("[ContainerManager]: Discovered runtime cgroups name", "cgroupsName", cont)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:509]:
```suggestion
			klog.V(2).InfoS("Discovered runtime cgroup name", "cgroupName", cont)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 8]
```diff
@@ -675,7 +675,7 @@ func (cm *containerManagerImpl) Start(node *v1.Node,
 			for _, cont := range cm.systemContainers {
 				if cont.ensureStateFunc != nil {
 					if err := cont.ensureStateFunc(cont.manager); err != nil {
-						klog.Warningf("[ContainerManager] Failed to ensure state of %q: %v", cont.name, err)
+						klog.InfoS("[ContainerManager] Failed to ensure state of", "ensureState", cont.name, "err", err)
```
[the remark this answers is not in the document]
Comment (ehashman) [reply pkg/kubelet/cm/container_manager_linux.go:678]: Warning -> Info unless we fail (which we don't here)

Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:678]:
```suggestion
						klog.InfoS("Failed to ensure state", "containerName", cont.name, "err", err)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 9]
```diff
@@ -825,12 +825,12 @@ func isProcessRunningInHost(pid int) (bool, error) {
 	if err != nil {
 		return false, fmt.Errorf("failed to find pid namespace of init process")
 	}
-	klog.V(10).Infof("init pid ns is %q", initPidNs)
+	klog.V(10).InfoS("Init pid ns is", "initPidNs", initPidNs)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:828]:
```suggestion
	klog.V(10).InfoS("Found init PID namespace", "namespace", initPidNs)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 10]
```diff
@@ -825,12 +825,12 @@ func isProcessRunningInHost(pid int) (bool, error) {
...
 	processPidNs, err := os.Readlink(fmt.Sprintf("/proc/%d/ns/pid", pid))
 	if err != nil {
 		return false, fmt.Errorf("failed to find pid namespace of process %q", pid)
 	}
-	klog.V(10).Infof("Pid %d pid ns is %q", pid, processPidNs)
+	klog.V(10).InfoS("Pid", pid, "pidNs", processPidNs)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:833]:
```suggestion
	klog.V(10).InfoS("Process info", "pid", pid, "namespace", processPidNs)
```

File: pkg/kubelet/cm/container_manager_linux.go (hunk 2 above)
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:875]:
```suggestion
	klog.V(4).InfoS("Unable to get pid from file", "path", pidFile, "err", err)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 11]
```diff
@@ -912,7 +912,7 @@ func ensureProcessInContainerWithOOMScore(pid int, oomScoreAdj int, manager cgro
 		return err
 	} else if !runningInHost {
 		// Process is running inside a container. Don't touch that.
-		klog.V(2).Infof("pid %d is not running in the host namespaces", pid)
+		klog.V(2).InfoS("Pid is not running in the host namespaces", "pid", pid)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:915]:
```suggestion
		klog.V(2).InfoS("PID is not running in the host namespace", "pid", pid)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 12]
```diff
@@ -941,9 +941,9 @@ func ensureProcessInContainerWithOOMScore(pid int, oomScoreAdj int, manager cgro
 
 	// Also apply oom-score-adj to processes
 	oomAdjuster := oom.NewOOMAdjuster()
-	klog.V(5).Infof("attempting to apply oom_score_adj of %d to pid %d", oomScoreAdj, pid)
+	klog.V(5).InfoS("Attempting to apply oom_score_adj of to pid", "oomScoreAdj", oomScoreAdj, "pid", pid)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:944]:
```suggestion
	klog.V(5).InfoS("Attempting to apply oom_score_adj to process", "oomScoreAdj", oomScoreAdj, "pid", pid)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 13]
```diff
@@ -941,9 +941,9 @@ func ensureProcessInContainerWithOOMScore(pid int, oomScoreAdj int, manager cgro
...
 	if err := oomAdjuster.ApplyOOMScoreAdj(pid, oomScoreAdj); err != nil {
-		klog.V(3).Infof("Failed to apply oom_score_adj %d for pid %d: %v", oomScoreAdj, pid, err)
+		klog.V(3).InfoS("Failed to apply oom_score_adj for pid", "oomScoreAdj", oomScoreAdj, "pid", pid, "err", err)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:946]:
```suggestion
		klog.V(3).InfoS("Failed to apply oom_score_adj to process", "oomScoreAdj", oomScoreAdj, "pid", pid, "err", err)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 14]
```diff
@@ -991,10 +991,10 @@ func getContainer(pid int) (string, error) {
 	// in addition, you would not get memory or cpu accounting for the runtime unless accounting was enabled on its unit (or globally).
 	if systemd, found := cgs["name=systemd"]; found {
 		if systemd != cpu {
-			klog.Warningf("CPUAccounting not enabled for pid: %d", pid)
+			klog.InfoS("CPUAccounting not enabled for pid", "pid", pid)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:994]:
```suggestion
			klog.InfoS("CPUAccounting not enabled for process", "pid", pid)
```

File: pkg/kubelet/cm/container_manager_linux.go [hunk 15]
```diff
@@ -991,10 +991,10 @@ func getContainer(pid int) (string, error) {
...
 		}
 		if systemd != memory {
-			klog.Warningf("MemoryAccounting not enabled for pid: %d", pid)
+			klog.InfoS("MemoryAccounting not enabled for pid", "pid", pid)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:997]:
```suggestion
			klog.InfoS("MemoryAccounting not enabled for process", "pid", pid)
```

Note: PID = process identifier. I think it's confusing to have "unable to x for pid pid=NNNN" so I suggest we replace pid with process in the message field

File: pkg/kubelet/cm/container_manager_linux.go [hunk 16]
```diff
@@ -1034,7 +1034,7 @@ func ensureSystemCgroups(rootCgroupPath string, manager cgroups.Manager) error {
 			return nil
 		}
 
-		klog.Infof("Moving non-kernel processes: %v", pids)
+		klog.InfoS("Moving non-kernel processes", "processes", pids)
```
Comment (ehashman) [inline pkg/kubelet/cm/container_manager_linux.go:1037]:
```suggestion
		klog.InfoS("Moving non-kernel processes", "pids", pids)
```

**Review by ehashman — COMMENTED** [review]
/ok-to-test
/remove-priority backlog
/priority important-soon

File: pkg/kubelet/cm/container_manager_linux.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (shiyajuan123) [reply pkg/kubelet/cm/container_manager_linux.go:678]: can you elaborate it,please?

File: pkg/kubelet/cm/container_manager_linux.go [hunk 17]
```diff
@@ -500,13 +500,13 @@ func (cm *containerManagerImpl) setupNode(activePods ActivePodsFunc) error {
...
-			klog.V(4).Infof("[ContainerManager]: Adding periodic tasks for docker CRI integration")
+			klog.V(4).InfoS("Adding periodic tasks for docker CRI integration")
 			cont, err := getContainerNameForProcess(dockerProcessName, dockerPidFile)
 			if err != nil {
-				klog.Error(err)
+				klog.ErrorS(err, "Failed to get container name for process")
 				return
 			}
-			klog.V(2).Infof("[ContainerManager]: Discovered runtime cgroups name: %s", cont)
+			klog.V(2).InfoS("Discovered runtime cgroup name", "cgroupName", cont
```
Comment (yuzhiquan) [inline pkg/kubelet/cm/container_manager_linux.go:509]: losing ")"

**Review by kikisdeliveryservice — COMMENTED** [review]
Went through outstanding comments and verified all requested changes were made.

/lgtm

**Review by mrunalp — APPROVED** [review]

**Pull request merged**